### PR TITLE
feat(gov): Implement gov slice

### LINKED ISSUE #894: feat: Implement gov slice
- Implement govService
- Implement govSlice

### PR DESCRIPTION
**Pull request by Hemanthghs**
Closes #894

### CODE REVIEW

File: frontend/src/types/gov.d.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  deposit_end_time: string;
+  total_deposit: {
+    denom: string;
+    amount: string;
+  }[];
+  voting_start_time: string;
+  voting_end_time: string;
+}
+
+interface GovPagination {
```
Comment (aleem1314) [inline frontend/src/types/gov.d.ts:27]: Pagination response is same for all APIs. We should move this to global types.

**Review by aleem1314 — COMMENTED** [review]
Looks good to me. We should also add support for gov v1 types. Let's create a follow up issue to add that.

File: frontend/src/store/features/gov/govSlice.ts [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+  async (data: GetProposalsInVotingInputs, { rejectWithValue, dispatch }) => {
+    try {
+      const response = await govService.proposals(
+        data.baseURL,
+        data.key,
+        data.limit,
+        PROPOSAL_STATUS_ACTIVE
+      );
+
+      if (response?.data?.proposals?.length) {
```
Comment (charymalloju) [inline frontend/src/store/features/gov/govSlice.ts:212]: its better to write to like this 
```suggestion
     const { data } = response || {};
const proposals = data?.proposals || [];

proposals.forEach((proposal) => {
  const proposalId = Number(proposal.proposal_id);

  dispatch(getProposalTally({
    baseURL: data?.baseURL,
    proposalId,
    chainID: data?.chainID,
  }));

  dispatch(getVotes({
    baseURL: data?.baseURL,
    proposalId,
    voter: data?.voter,
    chainID: data?.chainID,
  }));
});

```

File: frontend/src/store/features/gov/govSlice.ts [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+    { rejectWithValue, fulfillWithValue, dispatch }
+  ) => {
+    try {
+      const msg = GovVoteMsg(data.proposalId, data.voter, data.option);
+      const result = await signAndBroadcast(
+        data.chainID,
+        data.aminoConfig,
+        data.prefix,
+        [msg],
+        860000,
```
Comment (charymalloju) [inline frontend/src/store/features/gov/govSlice.ts:305]: define values in constants

File: frontend/src/store/features/gov/govSlice.ts [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+        data?.justification || '',
+        `${data.feeAmount}${data.denom}`,
+        data.rest,
+        data.feegranter?.length > 0 ? data.feegranter : undefined
+      );
+      if (result?.code === 0) {
```
Comment (charymalloju) [inline frontend/src/store/features/gov/govSlice.ts:373]:
```suggestion
     const { code, transactionHash, rawLog } = result || {};

if (code === 0) {
  dispatch(setTxHash({ hash: transactionHash }));
  return fulfillWithValue({ txHash: transactionHash });
} else {
  dispatch(setError({ type: 'error', message: rawLog || '' }));
  return rejectWithValue(rawLog);
}
```

**Review by charymalloju — APPROVED** [review]

**Pull request merged**